backend/src/main/java/com/example/cms/page/PageSpecification.java

```diff
@@ -29,49 +22,52 @@ public class PageSpecification implements Specification<Page> {
 
     @Override
     public Predicate toPredicate(Root<Page> root, CriteriaQuery<?> query, CriteriaBuilder criteriaBuilder) {
-        Predicate predicates = criteriaBuilder.and(criteriaBuilder.or(
-                criteriaBuilder.and(
-                        criteriaBuilder.equal(root.get("hidden"), false),
-                        criteriaBuilder.equal(root.get("university").get("hidden"), false)
-                )));
-
-        if (this.role != null) {
-            predicates = criteriaBuilder.and(predicates,
-                    criteriaBuilder.equal(criteriaBuilder.literal(role), "ADMIN"));
+//        Predicate predicates = criteriaBuilder.and(criteriaBuilder.or(
```

Comment:
Nie działa tak jak powinno, gdy użytkownik jest zalogowany to nie działa filtrowanie po `University.id` ani `Creator.id`, w dodatku `CompoundSpecification` jest jakieś dziwnie wielkie (masa `and` w wygenerowanym sqlu pomimo posiadania tylko 3 warunków).

To jest do rozpatrzenia i naprawy @WiktorSzymanski

Reply:
Utworzyłem issue  #36 pod to

Reply:
To teraz mam sprawdzać jak jest czy czekać na merge #36 i wtedy sprawdzić to?

Reply:
Możesz zmergowac najpierw swoje

Reply:
Wycofałem zmiany, które konfliktują z #36 więc w sumie możesz sprawdzać :)